### PR TITLE
WT-7864 Add support to run.py for running lists/ranges of scenarios in a test

### PR DESCRIPTION
**Pull request by sauclovian-wt**
Add support to run.py for running lists/ranges of scenarios with -s.

The syntax supported matches e.g. 1-3,5,7-9. A single number is handled
as a trivial list.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by luke-pearson** [comment]
Thanks for submitting this improvement, I'll review it today. 👍

File: test/suite/run.py [hunk 1]
```diff
@@ -178,16 +178,38 @@ def show_env(verbose, envvar):
 # e.g. test_util03 -> util
 reCatname = re.compile(r"test_([^0-9]+)[0-9]*")
 
+# look for a list of the form 0-9,11,15-17
```
Comment (luke-pearson) [inline test/suite/run.py:181]: Capitalize the start of comments.

File: test/suite/run.py [hunk 2]
```diff
@@ -178,16 +178,38 @@ def show_env(verbose, envvar):
...
+def parse_int_list(str):
+    ret = {}
+    # divide into ranges separated by commas
+    for r in str.split(","):
+        # split the range
+        bounds = r.split("-")
+        if len(bounds) == 1 and bounds[0].isdigit():
+            # single number with no dash
+            scenario = int(bounds[0])
+            ret[scenario] = True
```
Comment (luke-pearson) [inline test/suite/run.py:191]: The choice to use a dictionary here is to speed up the lookup later on, correct?

**Comment by luke-pearson** [comment]
Please update the usage text on line 134.

File: test/suite/run.py (hunk 1 above)
Comment (sauclovian-wt) [reply test/suite/run.py:181]: Fixed. Sorry, going to take a while to get used to that...

File: test/suite/run.py (hunk 2 above)
Comment (sauclovian-wt) [reply test/suite/run.py:191]: Yes. I mean, it doesn't really matter, but might as well.

**Comment by sauclovian-wt** [comment]
Updated the usage message too, good point :-)

File: test/suite/run.py [hunk 3]
```diff
@@ -178,16 +179,38 @@ def show_env(verbose, envvar):
 # e.g. test_util03 -> util
 reCatname = re.compile(r"test_([^0-9]+)[0-9]*")
 
+# Look for a list of the form 0-9,11,15-17.
+def parse_int_list(str):
+    ret = {}
+    # divide into ranges separated by commas
```
Comment (sueloverso) [inline test/suite/run.py:185]: @sauclovian-wt all comments should be fully formed sentences.

Comment (sauclovian-wt) [reply test/suite/run.py]: Right, I fixed the rest of them. Sorry, this is probably going to continue to happen some in the short term as I'm not used to that requirement :-/

File: test/suite/run.py (hunk 2 above)
Comment (luke-pearson) [reply test/suite/run.py:192]: Can you add a comment on line 183 stating the reason behind that choice? It's fairly obvious but I'd like it be explicit.

**Comment by luke-pearson** [comment]
@sauclovian-wt This is looking good to me, one final comment regarding adding an additional comment 👍 .

File: test/suite/run.py (hunk 2 above)
Comment (sauclovian-wt) [reply test/suite/run.py:192]: Sure.

Comment (luke-pearson) [reply test/suite/run.py:194]: Thanks, that's what I expected.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Review by sueloverso — APPROVED** [review]
LGTM

**Comment by sauclovian-wt** [comment]
Thanks!

**Pull request merged**